.circleci/config.yml

```diff
@@ -1255,10 +1332,15 @@ workflows:
     jobs:
       - lint
       - run-test-ios-17
+      - run-test-ios-18
       - pod-lib-lint
       - run-revenuecat-ui-ios-17
+      - run-revenuecat-ui-ios-18
 
   create-tag:
+    executor:
+      name: macos-executor
+      xcode_version: '15.2.0'
```

Comment:
does it not work with Xcode 16?

Reply:
No :(

The fastlane `pod_lib_lint` job fails because of the Sendable warnings that we've been facing. `pod lib lint` doesn't provide a way to silence specific warnings, it's either all or nothing, so I decided to continue using Xcode 15 for now to avoid those warnings but still see any other warnings that might pop up.

Here are the warnings we're seeing:
![image](https://github.com/user-attachments/assets/dcf8e4f4-1bb9-4a41-a4bb-409b748d817e)

We should definitely address them sooner or later, but this PR is already big enough so I didn't want to add them here.

Reply:
Same applies to the `pod-lib-lint` job too! https://github.com/RevenueCat/purchases-ios/pull/4295#discussion_r1782936624

Reply:
I also just added comments here explaining why they're using Xcode 15.2.0 :D